plasma_framework/contracts/src/exits/payment/controllers/PaymentChallengeIFENotCanonical.sol

```diff
@@ -74,6 +74,8 @@ library PaymentChallengeIFENotCanonical {
     )
         public
     {
+        require(args.senderData == keccak256(abi.encodePacked(msg.sender)), "Incorrect senderData");
```

Comment:
doesn't the "respond to non-canonical challenge" (not sure of the exact name) require one as well?

Reply:
Hmm, good point. But maybe neither "challengeInFlightExitNotCanonical" nor "respondToNonCanonicalChallenge" need this protection right now.

That's because the IFE bond doesn't get paid out on a canonical challenge or response - the _owner_ of the bond is changed here, but the bond doesn't get paid out until processExit is called.

So while a front runner could potentially benefit by front running a canonical challenge or response, they would have to be very familiar with how the exit game works.

Um. I'm inclined to add it in anyway though, for consistency with the other methods, and possible an easier path to adding more sophisticated front running protection later.

What do you think?

Reply:
ah you're right! I'd definitely vote for consistent, to not surprise our future selves reading the code.

Just the question now is - what will be consistent:
 - `senderAddress` on anything resembling a challenge and related to any bond operation or
 - `senderAddress` on anything that actually can pay anything out.

On 2nd thought, the latter makes more sense, so it seems it should be dropped from `challengeIFENotCanon.`. The downside I see is that if sometime, someone decides that those call should pay some portion of the bond (for whatever reason), they'd have to remember to do `senderAddress` check.

Not 100% sure here.

Reply:
I'm going with option 2; only putting senderData on anything that actually pays out.
Trusting our future selves to do the right thing if or when we implement a more sophisticated front running solution